src/objects/DateRange.tsx

```diff
@@ -0,0 +1,190 @@
+import { DateRangeUI } from "./Types";
+import { Locale } from "./Locale";
+import { MSFormatter } from "./MSFormatter";
+
+export enum DateType {
+  absolute = "absolute",
+  relative = "relative",
+}
+
+enum DateIndex {
+  start = 0,
+  end = 1,
+}
+
+export enum TermContext {
+  quickSelect = "quickSelect",
+  relative = "relative",
+}
+
+export class DateRange {
```

Comment:
Still haven’t decomposed this, I see :)

Reply:
you said i could in the next PR. This ones getting pretty huge